### PR TITLE
EXSWHTEC-265 - Implement Tests for Atomic Exchange

### PR DESCRIPTION
**Pull request by music-dino**
- Implement positive tests for combinations of a single kernel on a single device and concurrent kernels on a single device with various memory access patterns for atomicExch
 - Implement positive tests for combinations of a single kernel on a single device concurrent with host,  and concurrent kernels on  multiple devices concurrent with host, with various memory access patterns for atomicExch_system
 - Implement negative parameter tests for atomicExch and atomicExch_system.

### CODE REVIEW

**Review by b-sumner — APPROVED** [review]
LGTM

**Comment by rorake** [comment]
@music-dino 
Will you please resolve the merge conflicts for this PR?

**Comment by rakesroy** [comment]
@music-dino 
Due to recent pushes to develop branch, merge conflicts occurred in this PR.
Will you please resolve them?

File: catch/include/hip_test_defgroups.hh [hunk 1]
```diff
@@ -95,8 +97,46 @@ THE SOFTWARE.
...
+ *    -# Utilizes sub-process to invoke compilation of faulty source.
+ *    -# Performs post-processing of compiler output and counts errors.
+ * Test source
+ * ------------------------
+ *  - unit/atomics/CMakeLists.txt
+ * Test requirements
+ * ------------------------
+ *  - HIP_VERSION >= 5.2
+ */
+TEST_CASE("Unit_atomicExch_Negative_Parameters") {}
```
Comment (rakesroy) [inline catch/include/hip_test_defgroups.hh:132]: @mirza-halilcevic, due to this line we are getting multiple definition error:
multiple_definition_error

Will you please check?

**Comment by mangupta** [comment]
PR has been merged into develop branch via commit f24d2dd934f2438d31dce2c73a89c416b915df6d.

**Pull request closed without merging**